backend/src/main/java/com/bakdata/conquery/util/search/TrieSearch.java

```diff
@@ -0,0 +1,187 @@
+package com.bakdata.conquery.util.search;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.IntSummaryStatistics;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import com.google.common.base.Strings;
+import it.unimi.dsi.fastutil.objects.Object2DoubleAVLTreeMap;
+import it.unimi.dsi.fastutil.objects.Object2DoubleMap;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.trie.PatriciaTrie;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Trie based keyword search for autocompletion and resolving.
+ *
+ * We store not only whole words but suffixes up to length of SUFFIX_CUTOFF to enable a sort of fuzzy and partial search with longer and compound search terms.
+ */
```

Comment:
Wollen wir nicht eigentlich Prefixes ab einer bestimmten Länge?
Also so wie Gitlab das macht: Erst Ergebisse liefern ab 3 Zeichen

Reply:
Es läuft auf das gleiche hinaus: Wir bauen hier einen Index über alle Suffixe bis länge 3 vom Start ausgehen.
Also Jamaica wird dann aufgesplittet in Jamaica, amaica, maica, aica, ica, und kann dann auch über alle Splits gesucht werden.

Reply:
Ah passt